src/components/common/button/Button.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { CSSProperties } from 'react';
+import { css } from '@emotion/react';
+import styled from '@emotion/styled';
+
+const Container = styled.div<{ size: ButtonSize; type: ButtonType }>`
+  display: flex;
+  height: 60px;
+  justify-content: center;
+  align-items: center;
+  color: #ffffff;
+  border-radius: 16px;
+
+  cursor: pointer;
+
+  ${({ type }) => css`
+    background-color: ${backgroundColors[type]};
+    &:hover,
+    &:focus {
+      background-color: ${hoverColors[type]};
+    }
+  `}
+
+  ${({ size }) => css`
+    width: ${buttonWidths[size]}px;
+    font-size: ${textSizes[size]}rem;
+  `}
+`;
+
+export enum ButtonSize {
+  XLARGE,
+  LARGE,
+  MEDIUM,
+  SMALL,
+}
+
+export enum ButtonType {
+  MINT,
+  GRAY,
+  DARKGRAY,
+  BLACK,
+  YELLOW,
+}
+
+const buttonWidths = {
+  [ButtonSize.XLARGE]: 480,
+  [ButtonSize.LARGE]: 320,
+  [ButtonSize.MEDIUM]: 240,
+  [ButtonSize.SMALL]: 120,
+};
+
+const textSizes = {
+  [ButtonSize.XLARGE]: 1.5,
+  [ButtonSize.LARGE]: 1.5,
+  [ButtonSize.MEDIUM]: 1.5,
+  [ButtonSize.SMALL]: 1,
+};
+
+const backgroundColors = {
+  [ButtonType.MINT]: '#4EC1BF',
+  [ButtonType.GRAY]: '#B7B7B7',
+  [ButtonType.DARKGRAY]: '#707070',
+  [ButtonType.BLACK]: '#2F2F2F',
+  [ButtonType.YELLOW]: '#FFD540',
+};
+
+const hoverColors = {
+  [ButtonType.MINT]: '#35D8D5',
+  [ButtonType.GRAY]: '#B7B7B7',
+  [ButtonType.DARKGRAY]: '#707070',
+  [ButtonType.BLACK]: '#2F2F2F',
+  [ButtonType.YELLOW]: '#FFD540',
+};
+
+interface IButton {
+  size?: ButtonSize;
+  type?: ButtonType;
+  containerStyle?: CSSProperties;
+  onClick?: (v: any) => any;
+  children: React.ReactNode;
+}
+
+export default function Button({
+  size = ButtonSize.XLARGE,
+  type = ButtonType.MINT,
+  containerStyle,
+  onClick,
+  children,
+}: IButton) {
+  return (
+    <Container size={size} type={type} style={containerStyle} onClick={onClick}>
+      {children}
+    </Container>
+  );
+}
```

Comment:
button의 html 속성을 extend 하면 속성을 일일이 지정해 주지 않아도 가능합니다!

```suggestion
interface IButton extends HTMLAttributes<HTMLButtonElement> {
  size?: ButtonSize;
  type?: ButtonType;
  children: React.ReactNode;
}

export default function Button({
  size = ButtonSize.XLARGE,
  type = ButtonType.MINT,
  children,
  ...props
}: IButton) {
  return (
    <Container size={size} type={type} {...props}>
      {children}
    </Container>
  );
}
```